packages/mdx-v2-migrate/src/migrate.js

```diff
@@ -0,0 +1,127 @@
+  return remark().data('settings', {
+    bullet: '-',
+    emphasis: '_',
+    unsafe: [
+      { character: '>', before: '<', inConstruct: 'phrasing' },
+      { character: '<', after: '[a-zA-Z0-9]', inConstruct: 'phrasing' },
+      { character: '<', after: '[a-zA-Z0-9]', inConstruct: 'phrasing' },
+      // brackets are used to denote a JS expression, need to escape all current usages
+      { character: '{', after: '{', inConstruct: 'phrasing' },
+      { character: '{', after: '[a-zA-Z0-9]', inConstruct: 'phrasing' },
+    ],
+    listItemIndent: 'one',
+    incrementListMarker: false,
+    rule: '-',
+    resourceLink: true,
+    fences: true,
+    handlers: {
+      /**
+       * Rewrites HTML comments into MDX comments
+       *
+       * <!-- comment --> -> {/* comment *\/}
+       */
+      html(node) {
+        const match = node.value.match(commentExpression)
+
+        if (match) {
+          return `{/* ${match[1]} */}`
+        }
+
+        return node.value
+      },
+      /**
+       * We're doing a lot of un-escaping here, which seems to be fine as most of the valid markdown
+       * nodes have already been parsed and are no longer *text* nodes. The goal is to have the resulting
+       * re-generated MDX be as close to the original source as possible, we don't want to introduce escaping
+       * that might confuse authors if it's not necessary.
+       */
+      text(node, _, context, safeOptions) {
+        /**
+         * Unescape JSX tags that got escaped for unknown reasons (Not "valid" HTML look-a-likes?)
+         * and which are only in a phrasing node (i.e. they are most likely an opening JSX tag)
+         *
+         * match: \<CodeBlockConfig foo={{ foo: "bar" }}>
+         * no match: **\<foo>**
+         */
+        let safe = text(node, _, context, safeOptions)
+        if (
+          (safe.match(/^\\<\w.*?>$/m) || safe.match(/\\<\w.*?\/>/)) &&
+          ['phrasing', 'emphasis', 'strong'].includes(
+            context.stack.slice(-1)[0]
+          )
+        ) {
+          return node.value
+        }
+        // ensure our custom aside shorthand does not get escaped
+        if (safe.includes('\\->')) {
+          safe = safe.replace('\\->', '->')
+        }
+        if (safe.includes('\\~>')) {
+          safe = safe.replace('\\~>', '~>')
+        }
+
+        // Unescape _ characters in text nodes, they render without issue
+        if (safe.includes('\\_')) {
+          safe = safe.replace(/\\_/g, '_')
+        }
+        // Unescape * characters in text nodes, they're not in a strong/emphasis node and so are safe
+        if (safe.includes('\\*')) {
+          safe = safe.replace(/\\\*/g, '*')
+        }
+        // Unescape [ characters in text nodes, they're not part of a link node and so are safe
+        if (safe.includes('\\[')) {
+          safe = safe.replace(/\\\[/g, '[')
+        }
+
+        // allow usage of <= sign
+        safe = safe.replace('<=', '\\<=')
```

Comment:
unescaping some of these seems quite dangerous. If it works it works, but especially `_`, `*`, and `[` will interfere at some point

Reply:
Potentially, yeah. It seems safe as they are in text nodes at this point, so we know they aren't part of a pair elsewhere in the tree, but I agree it might not hold up for everything. Testing out the script on our docs sites now, we'll see if we run into any issues 🤞